examples/flax/image-captioning/run_image_captioning_flax.py

```diff
@@ -0,0 +1,1485 @@
+                decoder_config = AutoConfig.from_pretrained(
+                    model_args.decoder_model_name_or_path, cache_dir=decoder_cache_dir
+                )
+            # Use specified decoder model type
+            elif model_args.decoder_model_type:
+                decoder_config = AutoConfig.for_model(model_args.decoder_model_type)
+                logger.warning("You are instantiating a new config instance from scratch for the decoder.")
+            else:
+                raise ValueError(
+                    "Decoder Config: if pretrained config or model location is not provided, `decoder_model_type` is required."
+                )
+
+            logger.info("Setting `config.is_decoder=True` and `config.add_cross_attention=True` for decoder_config")
+            decoder_config.is_decoder = True
+            decoder_config.add_cross_attention = True
+
+            config = config_class.from_encoder_decoder_configs(encoder_config, decoder_config)
+        # For self-contained model
+        else:
+            config = config_class()
+            logger.warning("You are instantiating a new config instance from scratch.")
+
+    decoder_start_token_id = getattr(config, "decoder_start_token_id", None)
+    if not decoder_start_token_id and getattr(config, "decoder", None):
+        decoder_start_token_id = getattr(config.decoder, "decoder_start_token_id", None)
+    bos_token_id = getattr(config, "bos_token_id", None)
+    if not bos_token_id and getattr(config, "decoder", None):
+        bos_token_id = getattr(config.decoder, "bos_token_id", None)
+    eos_token_id = getattr(config, "eos_token_id", None)
+    if not eos_token_id and getattr(config, "decoder", None):
+        eos_token_id = getattr(config.decoder, "eos_token_id", None)
+    pad_token_id = getattr(config, "pad_token_id", None)
+    if not pad_token_id and getattr(config, "decoder", None):
+        pad_token_id = getattr(config.decoder, "pad_token_id", None)
+
+    if decoder_start_token_id is None:
+        decoder_start_token_id = bos_token_id
+    if pad_token_id is None:
+        pad_token_id = eos_token_id
+
+    if getattr(config, "decoder", None):
+        config.decoder.decoder_start_token_id = decoder_start_token_id
+        config.decoder.bos_token_id = bos_token_id
+        config.decoder.eos_token_id = eos_token_id
+        config.decoder.pad_token_id = pad_token_id
+
+    # Set `encoder-decoder` (top-level) specific config (not always necessary, but can avoid generate() error sometimes)
+    config.decoder_start_token_id = decoder_start_token_id
+    config.bos_token_id = bos_token_id
+    config.eos_token_id = eos_token_id
+    config.pad_token_id = pad_token_id
+
+    if model_args.model_name_or_path:
+        model = FlaxAutoModelForVision2Seq.from_pretrained(
+            model_args.model_name_or_path, config=config, seed=training_args.seed, dtype=getattr(jnp, model_args.dtype)
+        )
+    else:
```

Comment:
This whole else block is way too complex too.

Reply:
Agree with @sgugger here

I'm actually in favor of **not** using `.from_encoder_decoder_pretrained` further down below, but I think it would be better to just use `.from_pretrained(...)` instead similar to how it's done in this seq2seq example: https://github.com/huggingface/transformers/blob/e68c3756fea7c811d02b8470539ae17ec3ec0e71/examples/pytorch/speech-recognition/run_speech_recognition_seq2seq.py#L327 . The user should be responsible for creating the model her/himself before-hand. We could add some explanation on how to do this in a README.md e.g.: https://github.com/huggingface/transformers/tree/master/examples/pytorch/speech-recognition#sequence-to-sequence

IMO there are just too many different use-cases to cover and we should stick to the easiest one. The way it's currently implemented would also not allow a user to continue fine-tuning or to evaluate from a fine-tuned VisionEncoderDecoder model since it can only be loaded via `from_encoder_decoder_pretrained(...)`. 

=> I would prefer to follow the approach here: https://github.com/huggingface/transformers/blob/e68c3756fea7c811d02b8470539ae17ec3ec0e71/examples/pytorch/speech-recognition/run_speech_recognition_seq2seq.py#L327

Reply:
Yeah, in a previous version I tried to keep both the possibilities of using `from_pretrained` and `from_encoder_decoder_pretrained `. Let's hear others opinions for this part (which use case to support)

Reply:
I agree with what Patrick said. I would also prefer just using `from_pretrained`

Reply:
I add a new (small) script to allow users to create and save a model from specified encoder & decoder.

https://github.com/huggingface/transformers/blob/3a74bb15a68fbcd1d1fa83ac5150a27c0ca12e43/examples/flax/image-captioning/create_model_from_encoder_decoder_models.py#L91

And in `run_image_captioning_flax.py`, only support `from_pretrained` (so much cleaner too)

https://github.com/huggingface/transformers/blob/3a74bb15a68fbcd1d1fa83ac5150a27c0ca12e43/examples/flax/image-captioning/run_image_captioning_flax.py#L440-L452

Is this OK for you? (I will make sure it can run if you think this way is OK ). Thanks!

Reply:
Yes, I think that's a good idea! Let's also update the readme with an example of how to use this script, then we should be good. Thanks!